File: android/src/ti/airship/components/PushProxy.java
/* Copyright Airship and Contributors */

package ti.airship.components;

import androidx.core.app.NotificationManagerCompat;

import com.urbanairship.UAirship;

import org.appcelerator.kroll.KrollDict;
import org.appcelerator.kroll.KrollFunction;
import org.appcelerator.kroll.KrollProxy;
import org.appcelerator.kroll.annotations.Kroll;

import ti.airship.utils.PluginLogger;

@Kroll.proxy
public class PushProxy extends KrollProxy {

    @Kroll.method
    public KrollDict getNotificationStatus() {
        PluginLogger.logCall("getNotificationStatus");

        KrollDict dict = new KrollDict();
        dict.put("airshipOptIn", UAirship.shared().getPushManager().isOptIn());
        dict.put("airshipEnabled", UAirship.shared().getPushManager().getUserNotificationsEnabled());
        dict.put("systemEnabled", NotificationManagerCompat.from(getActivity()).areNotificationsEnabled());
        return dict;
    }

    @Kroll.method
    @Kroll.getProperty
    public String getPushToken() {
        return UAirship.shared().getPushManager().getPushToken();
    }

    @Kroll.method
    @Kroll.getProperty
    public boolean getUserNotificationsEnabled() {
        return UAirship.shared().getPushManager().getUserNotificationsEnabled();
    }

    @Kroll.method
    @Kroll.setProperty
    public void setUserNotificationsEnabled(boolean enabled) {
        PluginLogger.logCall("getUserNotificationsEnabled", enabled);
        UAirship.shared().getPushManager().setUserNotificationsEnabled(enabled);
    }

    @Kroll.method
    public void enableUserNotifications(KrollFunction function) {
        PluginLogger.logCall("enableUserNotifications", function);
        UAirship.shared().getPushManager().setUserNotificationsEnabled(true);
        if (function != null) {
            function.callAsync(getKrollObject(), new Boolean[] { UAirship.shared().getPushManager().isOptIn() });
        }
    }

}

File: android/src/ti/airship/components/PushProxy.kt
/* Copyright Airship and Contributors */
package ti.airship.components

import androidx.core.app.NotificationManagerCompat
import com.urbanairship.UAirship
import org.appcelerator.kroll.KrollDict
import org.appcelerator.kroll.KrollFunction
import org.appcelerator.kroll.KrollProxy
import org.appcelerator.kroll.annotations.Kroll.*
import ti.airship.utils.PluginLogger

@proxy
class PushProxy : KrollProxy() {
    @get:getProperty
    @get:method
    val notificationStatus: KrollDict
        get() {
            PluginLogger.logCall("getNotificationStatus")
            val dict = KrollDict()
            dict["airshipOptIn"] = UAirship.shared().pushManager.isOptIn
            dict["systemEnabled"] =
                NotificationManagerCompat.from(getActivity()).areNotificationsEnabled()
            return dict
        }

    @get:getProperty
    @get:method
    val pushToken: String?
        get() = UAirship.shared().pushManager.pushToken

    @get:getProperty
    @set:setProperty
    var userNotificationsEnabled: Boolean
        get() = UAirship.shared().pushManager.userNotificationsEnabled
        set(enabled) {
            PluginLogger.logCall("getUserNotificationsEnabled", enabled)
            UAirship.shared().pushManager.userNotificationsEnabled = enabled
        }

    @method
    fun enableUserNotifications(function: KrollFunction?) {
        PluginLogger.logCall("enableUserNotifications", function)
        UAirship.shared().pushManager.userNotificationsEnabled = true
        function?.callAsync(
            getKrollObject(),
            arrayOf(UAirship.shared().pushManager.isOptIn)
        )
    }
}